src/components/PeopleTable.tsx

```diff
@@ -1,644 +1,164 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
-  return (
+import { useParams, useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+import classNames from 'classnames';
+import { PersonLink } from './PersonLink';
+import { SearchLink } from './SearchLink';
+import { useMemo } from 'react';
+
+type Props = {
+  people: Person[];
+};
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { slug } = useParams();
+  const [searchParams] = useSearchParams();
+
+  const query = searchParams.get('query')?.toLowerCase() || '';
+  const gender = searchParams.get('sex');
+  const centuries = searchParams.getAll('centuries');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  const sortedOptions = ['Name', 'Sex', 'Born', 'Died'];
+
+  const filteredPeople = useMemo(() => {
+    let filtered = [...people];
+
+    if (query) {
+      filtered = filtered.filter(({ name, motherName, fatherName }) =>
+        [name, motherName, fatherName].some(field =>
+          field?.toLowerCase().includes(query)
+        )
+      );
+    }
+
+    if (gender) {
+      filtered = filtered.filter(person => person.sex === gender);
+    }
+
+    if (centuries.length > 0) {
+      filtered = filtered.filter(person => {
+        const century = Math.floor(person.born / 100) + 1;
+        return centuries.includes(century.toString());
+      });
+    }
+
+    return filtered;
+  }, [query, gender, centuries, people]);
+
+  const sortedPeople = useMemo(() => {
+    const sorted = [...filteredPeople].sort((personA, personB) => {
+      switch (sort) {
+        case 'name':
+        case 'sex':
+          return personA[sort].localeCompare(personB[sort]);
+        case 'born':
+        case 'died':
+          return personA[sort] - personB[sort];
+        default:
+          return 0;
+      }
+    });
+
+    if (order === 'desc') {
+      sorted.reverse();
+    }
+
+    return sorted;
+  }, [filteredPeople, sort, order]);
+
+  const toggleOrder = (sortName: string) => {
+    if (sort === sortName) {
+      return { sort: sortName, order: order === 'asc' ? 'desc' : 'asc' };
+    }
+    return { sort: sortName, order: 'asc' };
```

Comment:
The example page doesn't have order='asc', because it's default and we don't need it in the link. 

The third click (when people are already sorted in reversed order by this field) disables sorting;